cpp/src/c_api/neighbor_sampling.cpp

```diff
@@ -948,11 +948,13 @@ struct neighbor_sampling_functor : public cugraph::c_api::abstract_functor {
           std::exclusive_scan(
             recvcounts.begin(), recvcounts.end(), displacements.begin(), size_t{0});
 
-          (*label_to_comm_rank)
-            .resize(displacements.back() + recvcounts.back(), handle_.get_stream());
+          rmm::device_uvector<label_t> tmp_label_to_comm_rank(
```

Comment:
Why not just use `label_to_comm_rank.resize(...)` here?

Reply:
It is an optional which is initialized to `std::nullopt`

Reply:
```
label_to_comm_rank =
  rmm::device_uvector<label_t>(displacements.back() + recvcounts.back(), handle_.get_stream());
```
is sufficient (no need to create a temporary and move).